content/chat/rooms/moderation/lambda.textile

```diff
@@ -0,0 +1,101 @@
+---
+title: AWS Lambda
+meta_description: "Detect and remove unwanted content in a Chat Room using AWS Lambda."
+product: chat
+languages:
+  - javascript
+  - react
+  - swift
+  - kotlin
+---
+
+The AWS Lambda rule is a powerful way to bring your own moderation solution to Ably Chat. It allows you to run custom moderation logic or integrate with your preferred moderation provider by configuring an AWS Lambda function that will be invoked before messages are published to a chat room.
+
+This rule is particularly useful when you want to:
+* Integrate with a custom moderation service
+* Implement your own moderation logic
+* Use a moderation provider that isn't directly supported by Ably
+
+h2(#fields). Rule fields
+
+- AWS Region := The region where your Lambda function is deployed.
+- Function Name := The name of your AWS Lambda function.
+- AWS Authentication Scheme := The authentication method to use. Either @AWS Credentials@ or @ARN of an assumable role@. See the "Ably AWS authentication documentation":https://ably.com/docs/integrations/webhooks/lambda#aws-authentication.
+- Retry Timeout := Maximum duration (in milliseconds) that an attempt to invoke the rule may take (including any retries).
+- Max Retries := Maximum number of retries after the first attempt at invoking the rule.
+- Failed Action := The action to take in the event that a rule fails to invoke. Options are reject the request or publish anyway.
+- Room Filter (Optional) := A regular expression to match to specific chat rooms.
+
+h2(#lambda_setup). Lambda Function Setup
+
+To use the AWS Lambda rule, you need to create a Lambda function that adheres to the following API contract:
+
+h3(#request). Request Format
+
+<pre><code class="json">{
+  "source": "string",
+  "appId": "string",
+  "roomId": "string",
+  "site": "string",
+  "ruleId": "string",
+  "message": {
+    "roomId": "string",
+    "clientId": "string",
+    "text": "string",
+    "metadata": {
+      "key": "any"
+    },
+    "headers": {
+      "key": "string"
+    }
+  }
+}</code></pre>
+
+h3(#response). Response Format
+
+Your Lambda function must return a response in the following format:
+
+<pre><code class="json">{
```

Comment:
For Lambda specifically, we have the special body with statusCode that lets us handle user-code level retries. I've sent you a slack message on this :)